### DISCUSSION TITLE
2.0rc5 The final countdown - is out!

### DISCUSSION THREAD

**Discussion by Gissio** [Gissio/radpro#85] [Announcements]
2.0rc5 is out!

Most important, Bosean FS-5000 support has been added.

It is now possible to select the display panel on the FNIRSI GC-01, which should improve display quality a lot.

Dose is now available in the secondary view of the average measurement mode, and the conversion factor range has been extended to 25-1600 cpm/µSv/h. It is also possible to log data offline with a 1 second time interval.

Experimenters can now set the HV PWM frequency and duty cycle through the data communications interface, giving them control over the tube's high voltage.

Minor improvements include: 12-hour time format, improved data mode on devices with data mode, improved battery level indicator on the Bosean FS-600/FS-1000, improved battery level readout, improved keyboard control scheme on the Bosean FS-1000, and a change in the data communications protocol which now uses "\r\n" to end lines and works better with serial port terminals.

Please share your comments.

**Comment by Gissio** [comment]
Can you share your experiences with the new GC-01 display panel menu?

**Reply by dc1rdb** [reply]
Hi Gissio,
First of all, thank you for your efforts releasing 2.0rc5.
My GC-01 is equipped with a CL24CG1045-40B display, but selecting ENH-TV0240B026 results in much better contrast and readability. Selecting GMT024-05 V2.0 gives marginally better contrast than the default setting. 

CL24CG1045-40B (default setting) selected:
![IMG_20240529_082854](https://github.com/Gissio/radpro/assets/9314230/0a510522-d68f-4704-9d78-2ca11b6681b0)


ENH-TV0240B026 selected:
![IMG_20240529_082438](https://github.com/Gissio/radpro/assets/9314230/b39b9ede-0893-4d0c-b5e8-b1c3ba057756)

In dusk and night mode, the contrast difference is negligible.

**Reply by Gissio** [reply]
On the photos, the ENH panel settings look *way* better.

What about other users? Including also the ones with a CL24CG1045-40B?

**Reply by MSPTechLab** [reply]
Hi Gissio,
First of all, I'd like to say thank you for creation of software for gc-01.
I have a device with CL24CG1045-40B and can confirm the same as **dc1rdb** , with  ENH-TV0240B026 panel contrast is much better.

**Reply by Gissio** [reply]
Great. What do people with ENH and GMT panels say?

**Reply by alfmck** [reply]
Tested with FS-5000. Everything works as it should. Menu and functions are the same as in GC-01 Connecting to Geigerlog works too. Thank You, Gissio, very much for this great work!

**Reply by Montecri** [reply]
Mine is an ENH, and selecting ENH gives a much better contrast when using "dusk" theme. On "day" theme the difference is negligible.

**Reply by Gissio** [reply]
Can you post a photo?

So the CL24CG1045-40B profile can be deleted. What about people with a GMT or other panel? I'm eager to hear your reports.

**Reply by Montecri** [reply]
Yes.

![IMG_20240529_195133.jpg](https://github.com/Gissio/radpro/assets/38574378/32a5683e-a48b-498f-a872-e9437a57bee3)

But the picture is not doing it justice, it looks much better here.

**Reply by Gissio** [reply]
IMHO, it looks really nice!

**Comment by Montecri** [comment]
Just installed it on my FS-5000 and it works great!!!

THANK YOU, man!

**Reply by Gissio** [reply]
Is your display panel an “FPC1-H24C159-01W”? Does the display look good?

**Reply by Montecri** [reply]
Yes, it is. However the display is a bit inferior to the GC-01 display, which has great visibility in most angles. FS-5000 display you need to look straight on to have a good contrast.

**Reply by Montecri** [reply]
And yes, despite the viewing angles issue, it looks good.

**Comment by Montecri** [comment]
BTW guys, which one (GC-01 and FS500) do you think has "better" hardware? If there is any distinction... They appear to be almost identical.

GC-01 display has much better viewing angles.

**Reply by Gissio** [reply]
There are significant differences in favor of the FS-5000:

* The closed-loop HV voltage regulator.
* The superior energy efficiency.
* More flash memory.

**Comment by Gissio** [comment]
On the FS-5000, can somebody compare power consumption of Rad Pro with the original firmware?

On the GC-01, can somebody with a GMT display panel tell us which display profile looks better?

**Reply by alfmck** [reply]
About FS-5000: Is it possible to program the display that a view on it be rotated by 90 degrees? I already used to have big digits on GC-01.

**Reply by alfmck** [reply]
Power consumption FS-5000:
original firmware: lcd on - 22.6mA, Lcd off - 5.5mA, 
long endurance mode lcd on - 14.3mA, lcd off - 1.7mA, when led blinks - 2 mA.

radpro: lcd on - 18.2 mA (brightness medium), (9.1 mA on very low brightness), lcd off - 3.7 mA.
 I used battery from my GC-01, measured millivoltages on 1 Om resistor, sequency connected to battery.
For measurements I had to come back to original firmware using file saved in backup.

**Reply by Gissio** [reply]
Thanks a lot. There seems to be room for improvement.

Regarding your first question: pull the repository, in platformio.ini, in the bosean-fs5000 section, change:

    -D DISPLAY_240X320

to:

    -D DISPLAY_320X240

**Reply by alfmck** [reply]
Made change to -D DISPLAY_320X240, but there isn't any change on the FS-5000. Maybe I had to copy from project to flashtools some more, not only radpro-bosean-fs5000-2.0rc5.bin?

**Reply by Gissio** [reply]
If nothing changes: could you have made a mistake somewhere?

**Reply by alfmck** [reply]
I fixed a mistake and got now such a view:
![IMG_3291](https://github.com/Gissio/radpro/assets/165254519/ca40997f-9a5f-41d5-a382-35aeb4b22938)

**Reply by Gissio** [reply]
Did you pull the repository?

    git pull

**Comment by Gissio** [comment]
Once again: on the GC-01, can somebody with a GMT display panel tell us which display profile looks better?

**Comment by DG0MG** [comment]
Only a photo.

![IMG_7490](https://github.com/Gissio/radpro/assets/121504683/8435064e-7f70-43b3-9001-859cb2b7493d)

**Comment by alfmck** [comment]
Everything works! Thank You!  Sent from my iPhoneOn 1 Jun 2024, at 15:26, Gissio ***@***.***> wrote:﻿
Did you pull the repository?
git pull

**Reply by Gissio** [reply]
Can you post photos?

Are the keys mapped correctly with the new screen orientation?

Would other users want this as well?

**Reply by alfmck** [reply]
![IMG_3294](https://github.com/Gissio/radpro/assets/165254519/4bf09036-ac9a-4fe3-96cc-c9adb96f8385)

**Reply by alfmck** [reply]
![IMG_3292](https://github.com/Gissio/radpro/assets/165254519/96bf63db-47e4-4237-b74b-2b42a6b94b90)
![IMG_3295](https://github.com/Gissio/radpro/assets/165254519/b6b5030a-610c-4351-998e-6258a8df0fdb)

**Reply by alfmck** [reply]
We noticed that in original firmware digits are of black color and they better seen on the sun. Radpro uses blue color and on sunny day is difficult to read

**Reply by TzOk83** [reply]
You can choose display colors from 3 predefined "themes".

**Reply by alfmck** [reply]
Yes, we can, but measured values are blue, not black.

**Reply by Gissio** [reply]
alfmck:

can you replace the corresponding lines in ~~`platform.io/src/stm32/gc01.c`~~ `platform.io/src/stm32/fs5000.c` with the following:

    MR_SEND_COMMAND(MR_ST7789_VCOMS),
    MR_SEND_DATA(0x36), // VCOM=1.45 V
    MR_SEND_COMMAND(MR_ST7789_VRHS),
    MR_SEND_DATA(0x12), // VRH=4.45 V

**Reply by alfmck** [reply]
In repository these commands already are in 154-157 lines. Maybe I have to make changes in rc5?

**Reply by Gissio** [reply]
I just realized I referred you to the wrong file (Iook above). Please try again.

**Reply by alfmck** [reply]
What I have to see? The measured values are darker, but still blue. Tomorrow I'll lend a device with original firmware and compare again both devices on the sunlight. Could be the fourth **sunlight** theme on display settings with black measured values?
![IMG_3300](https://github.com/Gissio/radpro/assets/165254519/5cd27955-c0d7-45b4-a0c2-a5eb8d3c8d4a)

**Reply by Gissio** [reply]
It looks a lot better!

**Reply by alfmck** [reply]
![IMG_3307](https://github.com/Gissio/radpro/assets/165254519/ba27265f-971f-4318-a1fe-f43b61234b01)
![IMG_3305](https://github.com/Gissio/radpro/assets/165254519/075b2421-4ae7-48ea-b787-7b46f4c258ce)
![IMG_3306](https://github.com/Gissio/radpro/assets/165254519/f5e30c31-f5d8-444f-b27c-379b2f51632c)
![IMG_3304](https://github.com/Gissio/radpro/assets/165254519/f97883f2-e07c-4884-a6e6-0d4ec22b959c)

**Reply by Gissio** [reply]
I've just updated the UI design system to [OpenBridge 5.0](https://www.openbridge.no/), which complies with the [WCAG 2.1](https://www.w3.org/WAI/standards-guidelines/wcag/glance/) standard.

Please pull the repository, and try again.

**Reply by alfmck** [reply]
I was unable to make a normal photo, but real picture looks great! When view is rotated, it's good visibility on changing viewing angle.

![IMG_3329](https://github.com/Gissio/radpro/assets/165254519/39832d31-6ac7-4e21-9ea3-0349bce70044)

**Reply by alfmck** [reply]
I compared again fs-5000 with original firmware to FS-5000 with new radpro version.
Fs-5000 with new radpro version uses light grey background. When changing viewing angle in two directions, the background color does not change, and we have a good looking when we use the view rotated by 90 degrees. When changing viewing angle in other two directions, the background changes from clear white in one direction to light black in other direction.
Original fs-5000 uses clear white background - in this case background color does not change when viewing in all directions.
https://filebin.net/hsim3ozlqpqgwqrv

**Reply by Gissio** [reply]
The difference is not due to the color scheme but the LCD initialization sequence.

I contacted the manufacturer of the LCD panel in the Bosean FS-5000, and they were so kind to supply the correct initialization sequence.

In fact, with this sequence, the display looks much better than with the initialization sequence used by Bosean, albeit the changes in viewing angle.

**Comment by MartinXO1** [comment]
Hey guys, newbie here - first of all, thank you Gissio for an absolutely awesome work. 
I have a question regarding data logging. I tried to find info on how to make it work on my GC-01, but I could't find anything.
When I enable it, it drains battery quite fast, but I cannot seem to find any file except blank readme.txt in the device memory. 
Could someone please clarify the data logging function for me ? It would be an awesome function to have.

**Reply by ricsim** [reply]
It depends on the Microcontroller. With some you need to enable the communication (WCH), with the other is is allways enabled.
Then you need GeigerLog in the RadPro edition, that will query the data via the serial port (or you send the commands yourself via a terminal program. The commands are documented in the developer documentation).

**Reply by MartinXO1** [reply]
Wow, thank you for such prompt reply. I have CH32F103 MC. I was hoping for a simple text file being generated and stored in internal memory, so I can export and process the data later. But I guess, the onboard memory is vastly limited for something like that.

**Reply by ricsim** [reply]
It DOES store it on the flash if you want.
And you CAN export it via serial. (geigerlog can make a CSV if you want.)
You can also use it online, to log the current values at the moment in geigerlog.
Both is possible.

**Reply by MartinXO1** [reply]
I tried to download geigerlog, but it seems to require python to run. I am absolute noob, when it comes to programming and such. I tried to open serial connection via PuTTY, since I am at least somewhat familiar with that, but since the device doesn't seem to communicate via commands, I am out of ideas. I just pretty much need to access whatever raw log file it makes and then I can process it myself. Is there any "explain it like I'm 5" way how to access the internal memory ?

**Reply by Gissio** [reply]
Please RTFM: https://github.com/Gissio/radpro/blob/main/docs/users.md#data-logging

Download the version linked there.

**Reply by MartinXO1** [reply]
Thank you ! I must have missed this version. Program runs, but now I have to figure out why it stops responding every time I try to connect the device. I tried restarting PC, running as admin, re-connecting geiger, setting COM port from AUTO to specific port, different order of enabling/disabling logging and data mode after/before connecting and same result - I press connect, program window goes gray with (not responding) at the top bar and thats it.

**Reply by alfmck** [reply]
I also have these hangs. I solved that by switch off the Bluetooth function in windows settings.

**Reply by MartinXO1** [reply]
alfmck, you're the best :) You are correct, after disabling BT, it works just fine. Just a bit inconvenient as without BT I lose keyboard, mouse and sound as most of my peripherials are on BT. Anyways, thanks to all of, guys. I know how annoying it can be to try to help someone, who has no idea about what he is doing, so I seriously appreciate the patience.

**Reply by alfmck** [reply]
After starting geigerlog and connecting to device you can switch on BT again. 
And once more - after getting a new /all history from your device and have a view on geigerlog, you should save history data to csv file. Then you could analyze data on Excel.
